protocol/communities/persistence.go

```diff
@@ -741,6 +741,24 @@ func (p *Persistence) CanceledRequestsToJoinForUser(pk string) ([]*RequestToJoin
 	return requests, nil
 }
 
+func (p *Persistence) CanceledRequestToJoinForUserForCommunityID(pk string, communityID []byte) (*RequestToJoin, error) {
+	row := p.db.QueryRow(`SELECT id,public_key,clock,ens_name,chat_id,community_id,state
+	FROM
+	communities_requests_to_join
+	WHERE
+	state = ? AND public_key = ? AND community_id = ?`,
+		RequestToJoinStateCanceled, pk, communityID)
+
+	request := &RequestToJoin{}
+
+	err := row.Scan(&request.ID, &request.PublicKey, &request.Clock, &request.ENSName, &request.ChatID, &request.CommunityID, &request.State)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, err
+	}
```

Comment:
I believe in case of `sql.ErrNoRows` we should return nil.

```suggestion
	if err == sql.ErrNoRows {
		return nil, nil
	} else if err != nil {
		return nil, err
	}
```